Sources/IdentityState.swift

```diff
@@ -236,6 +240,14 @@ class IdentityState {
         eventDispatcher(event)
     }
 
+    /// Save `identityProperties` to persistence and resolves the XDM shared state.
+    /// - Parameters:
+    ///   - resolveXDMSharedState: function which resolves the XDM shared state
+    private func saveToPersistence(and resolveXDMSharedState: ([String: Any]) -> Void) {
+        identityProperties.saveToPersistence()
+        resolveXDMSharedState(identityProperties.toXdmData())
+    }
+
     /// Save `identityProperties` to persistence and create an XDM shared state.
     /// - Parameters:
     ///   - createXDMSharedState: function which creates an XDM shared state
```

Comment:
The only change in this PR other than the changes from dev is that this function is added back: 
```
private func saveToPersistence(and createXDMSharedState: ([String: Any], Event) -> Void, using event: Event)
```
In dev the update identity APIs use pending shared state and resolver. If adid does not change when setAdvertisingIdentifier API, I think we should not create a pending shared state and resolve with the same data, but just create a new shared state when the value changed.

Reply:
I hope we dont see the same race condition for adId then in case setAdId and getAdId being called back to back.

Reply:
There is no getAdid, but you can call getIdentities. For the consent use-case the shared state is updated first, then the consent -> edge events are being dispatched, so the data should be up to date at processing time.